### ISSUE TITLE
`./sbt all` should produce a pdf version of the graffle

### ISSUE THREAD

**Issue by Idloj** [NetLogo/NetLogo#806]

**Comment by SethTisue** [comment]
I agree this is desirable. But OmniGraffle costs money, so in order for this to be possible, we would have to make the diagram using some other software.

Normally we don't like to check derived files into version control, but I guess we should make an exception in this case, at least until someone redoes the diagram using only free tools.

So here's a PDF: https://github.com/NetLogo/NetLogo-Headless/commit/764c325230fa309ec6f26069362911756f8cd830

@mrerrormessage, could you let me know if the headless graffle is out of date, and if so, what changes I should make?

**Comment by Idloj** [comment]
How did you export the pdf?

**Comment by mrerrormessage** [comment]
@SethTisue , I see `core.mode` instead of `core.model`, and the date should be updated, but it looks good other than that.

I have used dot graph / graphviz for generating dependency graphs in the past. While they don't necessarily have the same pizzazz as omnigraffle, they can capture all the essential features of the omnigraffle graph (including color and grouping), they're easy to generate, and they can generate images using open-source (and free as in beer) software.

**Comment by mrerrormessage** [comment]
I think the headless graffle is pretty much up-to-date, but it might be good to do a little re-grouping. Could you elaborate on the colors vs the placement of nodes in certain sections? For instance "core.prim" and "core.prim.etc" are in the "parser" section, but the core package, but are colored like "prim" and "prim.etc". I spent a few minutes yesterday and today converting the graph to dot, the image rendered is provided below. I think we could pipeline this into our build, but it would take a little bit of effort. Perhaps that would be alright if it was used by both headless and JVM NetLogo. At any rate, here's the image, colors could also be added, but since I was unclear about colors I left it without. Feedback appreciated.

![depend](https://cloud.githubusercontent.com/assets/143198/9061589/6685157e-3a82-11e5-8831-6595328462ca.png)

**Comment by SethTisue** [comment]
In general, the colors are just an added dimension to help you see some additional structure that the nodes and arrows only don't show. My favorite example of when this is really useful and valuable is in the full graffle for desktop NetLogo, the HubNet packages have their own color. This is really helpful for finding all of the HubNet stuff, if you care about HubNet, or for tuning it out, if you don't care about it. (Another example is how the 2D and 3D rendering packages are called out using color.)

In the headless graffle, I don't specifically recall why I thought it was helpful to give the prim packages their own color. It seems to be the only time in the headless graffle where the colors are actually doing something different than the dotted lines are doing. If that just ends up being more confusing than helpful, feel free to get rid of it in your version, or ask me to get rid of it in the OmniGraffle version. I guess I'm still the only one who can edit that version. I'm happy to make any requested changes, just open a ticket, say what you want, and assign the ticket to me.

I'm not opposed to cutting OmniGraffle out of the picture entirely, on any or all repos/branches. Your dot version looks good to me.

**Comment by Idloj** [comment]
Now that headless repo is merged into the GUI one, is this more desirable and going to happen?
